pavelib/database.py

```diff
@@ -1,104 +1,126 @@
 from pavelib.utils.timer import timed
-from pavelib.utils.envs import Env
+
+# Bokchoy db schema and data fixtures
+BOKCHOY_DB_FILES = [
+    'bok_choy_data_default.json',
+    'bok_choy_data_student_module_history.json',
+    'bok_choy_migrations_data_default.sql',
+    'bok_choy_migrations_data_student_module_history.sql',
+    'bok_choy_schema_default.sql',
+    'bok_choy_schema_student_module_history.sql'
+]
+
+# Output files from scripts/calculate-bokchoy-migrations.sh
+MIGRATION_OUTPUT_FILES = [
+    'bok_choy_default_migrations.yaml',
+    'bok_choy_student_module_history_migrations.yaml'
+]
+
+ALL_DB_FILES = BOKCHOY_DB_FILES + MIGRATION_OUTPUT_FILES
+CACHE_BUCKET_NAME = 'edx-tools-database-caches'
+CACHE_FOLDER = 'common/test/db_cache'
 
 
 @needs('pavelib.prereqs.install_prereqs')
 @PassthroughTask
 @timed
 def update_bokchoy_db_cache():
     """
-    Update and cache the MYSQL database for bokchoy testing. This command
-    will remove any previously cached database files and apply migrations
-    on a fresh db. Additionally, the collective sha1 checksum for all of
-    these files will be written to file, for future comparisons/checking
-    for updates.
-
-    You can commit the resulting files in common/test/db_cache into
-    git to speed up test runs
+    Update and cache the MYSQL database for bokchoy testing:
+    * Remove any previously cached database files
+    * Apply migrations on a fresh db
+    * Write the collective sha1 checksum for all of these files to disk
+
+    WARNING: This will take several minutes.
     """
-    bokchoy_db_files = [
-        'bok_choy_data_default.json',
-        'bok_choy_data_student_module_history.json',
-        'bok_choy_migrations_data_default.sql',
-        'bok_choy_migrations_data_student_module_history.sql',
-        'bok_choy_schema_default.sql',
-        'bok_choy_schema_student_module_history.sql'
-    ]
     print('Removing cached db files for bokchoy tests')
-    for db_file in bokchoy_db_files:
-        try:
-            db_file_path = os.path.join(
-                '{}/common/test/db_cache'.format(Env.REPO_ROOT), db_file
-            )
-            os.remove(db_file_path)
-            print('\tRemoved {}'.format(db_file_path))
-        except OSError:
-            continue
-
-    sh('{}/scripts/reset-test-db.sh'.format(Env.REPO_ROOT))
-
-    # Write the fingerprint of the database files to disk for use in future
-    # comparisons
-    fingerprint = fingerprint_bokchoy_db_files()
-    with open('common/test/db_cache/bokchoy_migrations.sha1', 'w') as fingerprint_file:
-        fingerprint_file.write(fingerprint)
-
-
-def compare_bokchoy_db_fingerprints():
+    remove_files_from_folder(BOKCHOY_DB_FILES, CACHE_FOLDER)
+    apply_migrations(BOKCHOY_DB_FILES, update_cache_files=True)
+    compute_fingerprint_and_write_to_disk(MIGRATION_OUTPUT_FILES, ALL_DB_FILES)
+
+
+@needs('pavelib.prereqs.install_prereqs')
+@PassthroughTask
+@timed
+def update_local_bokchoy_db_from_s3():
```

Comment:
nit: In the case that either the fingerprints are equivalent or the fingerprint is not in the bucket, we will not be using s3. I feel like this could be a more generalized paver command name

Reply:
I agree. I was thinking of calling it update_local_bokchoy_db_from_s3_if_possible or something lolz.
I propose we merge as-is and then once we sort out exactly when each of these commands is called, and from where, we can rename.